build/pr-pipeline.yml

```diff
@@ -83,6 +83,8 @@ stages:
           catch
           {}
           New-AzResourceGroup -Name "$(ResourceGroupName)" -Location "$(ResourceGroupRegion)" -Force
+  - template: ./jobs/cleanup-aad.yml
```

Comment:
>cleanup [](start = 21, length = 7)

I find it's a bit strange to be part of this stage, and not one below (aadTestEnvironment)
In my head it feels more natural to bundle together work with AAD and put cleanup right before we setup test environment.

Reply:
It needs to happen before the Template deployment. I'll see if it makes sense to move the setup step here too

Reply:
Could you add it to the aadTestEnvironment stage as a second template before the add template? That way all the aad steps would be together.

Reply:
I changed some of the dependencies so I now have this, I think this looks better:
![image](https://user-images.githubusercontent.com/197221/109741268-91ad8d80-7b81-11eb-8e6f-289d97cd793f.png)

Reply:
>It needs to happen before the Template deployment.

Any chance you can elaborate? I'm not sure what Template you talking about.

Looking on that picture, why not put Setup AAD test as job into Provision Environment stage? (and give it better name than setup :D )